Add documentation on updating deploy rotation (#449)
18F/identity-handbook: _articles/appdev-deploy-rotation.md

layout: article
category: "AppDev"
subcategory: "Oncall"
description: "Spreadsheet to track the AppDev Deployer"
---

> The App-Dev team maintains a rotation for engineers who deploy the app.

Reviewer: Should this be AppDev? Not sure what the convention is for that.

Each week a new engineer is assigned from the rotation.

> The rotation spreadsheet can be found [here](https://docs.google.com/spreadsheets/d/1nLxhwVh4EfxdmqsvdFByxTk0bNSTyWGGBEXLODjlk6U/edit#gid=0).

Reviewer 2: I hate having link text that's just `here`. (I think there's SEO or readibilit benefits to avoiding it as well) I would recommend:
```suggestion
See the [rotation spreadsheet](https://docs.google.com/spreadsheets/d/1nLxhwVh4EfxdmqsvdFByxTk0bNSTyWGGBEXLODjlk6U/edit#gid=0).
```

## Overriding the deployer for a given week

Occasionally a deployer will need to trade weeks with another deployer to provide coverage.

The deployer should find someone who is willing to switch a week the them in advance of their deploy week.

> Once someone is identified, add a note to the rotation saying who the overrides is for their weeks.

Reviewer: What's the process when someone is unexpectedly out for an extended period and doesn't have a chance to arrange a replacement? Do we take them out, shift everyone's week, then add them back in later? Do we announce on #login-appdev that we need coverage? Something else?

## Updating the roster

The roster is intended to be updated as people roll-on or roll-off of Login.gov.

When a new person is added to the roster the future assignments will be shifted.

> Make sure to post in `#login-add-dev` and alert the next deployer if that slot has changed.

Reviewer 2: typo
```suggestion
Make sure to post in `#login-app-dev` and alert the next deployer if that slot has changed.
```

> Make sure to post in `#login-add-dev` and alert the next deployer if that slot has changed.

Reviewer: ```suggestion
Make sure to post in `#login-appdev` and alert the next deployer if that slot has changed.
```

#### Adding to the rotation

> To add someone, first change the value of the cell for the current deployer to include the deployer's name instead of the Sheets function.

Reviewer: Suggest adding instructions: Copy cell with cmd-C or ctrl-C, then right click, Paste Special, Values Only (cmd-shift-V or ctrl-shift-V).

This will minimize disruptions for future assignments.

> Next, add the new person to the roster before the current deployer.

Reviewer: Is this so that they will come up after a full rotation?

Author: That is correct. The idea is to minimize disruption for the next people in the queue

The new person should appear in future deployer slots.

#### Remove from the rotation

To remove someone from the rotation, first change the value of the cell for the person in the slot right
before their next occurrence.
Do this by replacing the deployer's name with the function in the cell value.

> This will minimize disruptions for future assignments.

Reviewer: I thought we had to replace all past cells with text rather than functions to avoid unwanted changes. Can you add more explanation here?

And, out of scope for this doc, but maybe there's an easier way to keep track of this that doesn't require spreadsheet shenanigans?
